core/object/class_db.h

```diff
@@ -185,33 +185,63 @@ class ClassDB {
 	}
 
 	template <class T>
-	static void register_class(bool p_virtual = false) {
+	static void register_class(bool p_virtual = false, bool p_exposed = true) {
```

Comment:
I'm not sure we want/need to add `p_exposed` on the existing `ClassDB::register_*()` functions?

At the moment, unexposed classes actually don't need to get manually registered all, they'll get added to `ClassDB` when the first instance of one of them is created. However, if we do want the ability to manually register them, I think `ClassDB::register_internal_class()` and `GDREGISTER_INTERNAL_CLASS()` would be sufficient?